mindmeld/components/dialogue.py

```diff
@@ -796,21 +796,22 @@ def _validate(self, request, slot):
         """
         text = request.text
         entity_type = slot.entity
-        query = self._extract_query_features(text)
-
-        # payload format for entity feature extractors:
-        #     tuple(query (Query Object), list of entities, entity index)
-
-        # For slot filling, the user query will consist of the prompted entity.
-        # Hence `entity = [query]` and `entity_index = 0`.
-        formatted_payload = (query, [query], 0)
 
         extracted_feature = {}
         _resolved_value = {}
+        query = None
 
         if slot.default_eval:
             if entity_type in DEFAULT_SYS_ENTITIES:
                 # system entity validation - checks for presence of required system entity
+
+                entity_text = (
+                    str(request.entities[0]["value"][0]["value"])
+                    if request.entities
+                    else text
+                )
```

Comment:
Does this mean we HAVE to detect the non-contextual entity using the tagger which are common in such slot filling use-cases? From my experience, its extremely hard to detect these non-contextual queries just by the tagger, which is why it made sense to pass the whole query through the `_extract_query_features ` function.

Reply:
Yeah that's exactly what is happening. If the entities are not detected through the tagger, then the whole query is passed through the ``_extract_query_features``.